### ISSUE TITLE
Saving a view with the same source in multiple views

### ISSUE THREAD

**Issue by K-Meech** [mobie/mobie-viewer-fiji#473]
Say, we have 2 views that each display the same source, but with a different affine transformation. If these views are not exclusive, they can both be added into the viewer and displayed with no problems. 
But, when we want to save a view, this does create issues... Both of these affine transforms will then be saved to the same view, affecting the same source. This will create a combined transform that doesn't make any sense.

How should this be handled? Perhaps we just print a warning if you try and save a view where the same source is displayed multiple times with different transformations?

**Comment by K-Meech** [comment]
This is also an issue because ```sourceTransformers``` will remain in the ViewerManager as long as at least one of the sources they apply to remains. So if you add one view with a source + affine transform, then another with the same source + affine transform. Then removing one of those will not remove either affine sourceTransformer - so again they get combined and give a combined affine that doesn't make sense. (so they linger in the view like here: https://github.com/mobie/mobie-viewer-fiji/issues/471)

**Comment by martinschorb** [comment]
> This will create a combined transform that doesn't make any sense.

If the transform contains a proper `sourceNamesAfterTransform` statement and the respective `imageDisplay` (or additional transforms) entries  (or additional transforms) refer to these derived sources, there should not be a problem.

Isn't this how multiple views from the same source should anyway be set up?

**Comment by K-Meech** [comment]
You're right, that if they have ```sourceNamesAfterTransform``` this shouldn't be an issue. But this is not currently required in the spec, it's optional.

**Comment by constantinpape** [comment]
> This is also an issue because `sourceTransformers` will remain in the ViewerManager as long as at least one of the sources they apply to remains. So if you add one view with a source + affine transform, then another with the same source + affine transform. Then removing one of those will not remove either affine sourceTransformer - so again they get combined and give a combined affine that doesn't make sense. (so they linger in the view like here: #471)

I don't quite understand the problem you describe here, @K-Meech. Can you try to reformulate it? Also, will this be fixed if #471 gets fixed?

> You're right, that if they have `sourceNamesAfterTransform` this shouldn't be an issue. But this is not currently required in the spec, it's optional.

Indeed, this is currently optional. I would also rather keep it that way, otherwise the spec for not so complex views that just apply a single transform per source get pretty complex.

**Comment by K-Meech** [comment]
Yes - @constantinpape, this is more of a detail of how ```sourceTransformers``` are stored, and should be fixed with https://github.com/mobie/mobie-viewer-fiji/issues/471.  I just wanted to make a note so I remember!

The bigger issue is what to do if someone tries to save a View when you have the same source being displayed multiple times with different transforms (and no ```sourceNamesAfterTransform```). I think I will just add a warning for now, and not allow the view to be saved - as there is no way to accommodate this into one view.

**Comment by constantinpape** [comment]
> Yes - @constantinpape, this is more of a detail of how `sourceTransformers` are stored, and should be fixed with #471. I just wanted to make a note so I remember!

Ok, thanks for the clarification.

> The bigger issue is what to do if someone tries to save a View when you have the same source being displayed multiple times with different transforms (and no `sourceNamesAfterTransform`). I think I will just add a warning for now, and not allow the view to be saved - as there is no way to accommodate this into one view.

Another option would be to auto-generate a `sourceNamesAfterTransform` in this case. But if that's to complicated for now throwing a warning and not saving the view is also fine.

**Comment by K-Meech** [comment]
That's true! I'll see how hard it is to implement :)
